static/app/views/settings/organizationIntegrations/configureIntegration.tsx

```diff
@@ -462,7 +460,23 @@ function ConfigureIntegration({params, router, routes, location}: Props) {
         ]
       : [];
 
-    const allTabs = tabs.concat(codeOwnerTabs);
+    // User mappings are mutually exclusive with stacktrace linking
+    // and code owners, so only render the main settings tab and user mappings.
+    const userMappingTabs: Array<[Tab, string]> = hasUserMapping
+      ? [
+          ['repos', t('Settings')],
+          ['userMappings', t('User Mappings')],
+        ]
+      : [];
+
+    const allTabs = tabs
+      .concat(stackTraceLinkingTabs)
+      .concat(codeOwnerTabs)
+      .concat(userMappingTabs);
+
+    if (allTabs.length === 0) {
+      return renderMainTab();
+    }
```

Comment:
Moved this to the bottom and made it the default if no other tabs are available to render vs short-circuiting.